ui/apps/platform/src/Containers/BaseImages/BaseImagesModal.tsx

```diff
@@ -1,27 +1,120 @@
-import { Alert, Button, Flex, FlexItem, Modal, Title } from '@patternfly/react-core';
-import type { AxiosError } from 'axios';
+import { useFormik } from 'formik';
+import type { FormikHelpers } from 'formik';
+import * as yup from 'yup';
+import {
+    Alert,
+    Button,
+    Flex,
+    Form,
+    FormGroup,
+    FormHelperText,
+    HelperText,
+    HelperTextItem,
+    Modal,
+    TextInput,
+    Title,
+} from '@patternfly/react-core';
 
+import { addBaseImage } from 'services/BaseImagesService';
 import { getAxiosErrorMessage } from 'utils/responseErrorUtils';
+import useRestMutation from 'hooks/useRestMutation';
 
 export type BaseImagesModalProps = {
     isOpen: boolean;
     onClose: () => void;
-    onSave: () => void;
-    isSuccess: boolean;
-    isError: boolean;
-    isSubmitting: boolean;
-    error: AxiosError | null;
+    onSuccess?: () => void;
 };
 
-function BaseImagesModal({
-    isOpen,
-    onClose,
-    onSave,
-    isSuccess,
-    isError,
-    isSubmitting,
-    error,
-}: BaseImagesModalProps) {
+const validationSchema = yup.object({
+    baseImagePath: yup
+        .string()
+        .required('Base image path is required')
+        .test(
+            'has-colon',
+            'Base image path must include both repository and tag separated by ":"',
+            (value) => {
+                if (!value?.includes(':')) {
+                    return false;
+                }
+                const lastColonIndex = value.lastIndexOf(':');
+                const tagPattern = value.substring(lastColonIndex + 1);
+                // Tag pattern must not be empty
+                return tagPattern.length > 0;
+            }
+        ),
+});
```

Comment:
Wanted to add at least some basic validation. We can leave the proper validation to the Backend and display the errors in the Modal's Alert. Anything more is unnecessary complexity and probably redundant.